cpp/tests/c_api/mg_ecg_test.c

```diff
@@ -0,0 +1,155 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mg_test_utils.h" /* RUN_TEST */
+
+#include <cugraph_c/algorithms.h>
+#include <cugraph_c/graph.h>
+
+#include <math.h>
+
+typedef int32_t vertex_t;
+typedef int32_t edge_t;
+typedef float weight_t;
+
+int generic_ecg_test(const cugraph_resource_handle_t* handle,
+                     vertex_t* h_src,
+                     vertex_t* h_dst,
+                     weight_t* h_wgt,
+                     vertex_t* h_result,
+                     size_t num_vertices,
+                     size_t num_edges,
+                     double min_weight,
+                     size_t ensemble_size,
+                     size_t max_level,
+                     double threshold,
+                     double resolution,
+                     bool_t store_transposed)
+{
+  int test_ret_value = 0;
+
+  cugraph_error_code_t ret_code = CUGRAPH_SUCCESS;
+  cugraph_error_t* ret_error;
+
+  cugraph_graph_t* graph                           = NULL;
+  cugraph_hierarchical_clustering_result_t* result = NULL;
+
+  int rank = cugraph_resource_handle_get_rank(handle);
+
+  ret_code = create_mg_test_graph(
+    handle, h_src, h_dst, h_wgt, num_edges, store_transposed, FALSE, &graph, &ret_error);
+
+  TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "create_test_graph failed.");
+  TEST_ALWAYS_ASSERT(ret_code == CUGRAPH_SUCCESS, cugraph_error_message(ret_error));
+
+  cugraph_rng_state_t* rng_state;
+  ret_code = cugraph_rng_state_create(handle, rank, &rng_state, &ret_error);
+  TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, "rng_state create failed.");
+
+  ret_code = cugraph_ecg(handle,
+                         rng_state,
+                         graph,
+                         min_weight,
+                         ensemble_size,
+                         max_level,
+                         threshold,
+                         resolution,
+                         FALSE,
+                         &result,
+                         &ret_error);
+
+  TEST_ASSERT(test_ret_value, ret_code == CUGRAPH_SUCCESS, cugraph_error_message(ret_error));
+  TEST_ALWAYS_ASSERT(ret_code == CUGRAPH_SUCCESS, "cugraph_ecg failed.");
+
+  if (test_ret_value == 0) {
+    cugraph_type_erased_device_array_view_t* vertices;
+    cugraph_type_erased_device_array_view_t* clusters;
+
+    vertices = cugraph_hierarchical_clustering_result_get_vertices(result);
+    clusters = cugraph_hierarchical_clustering_result_get_clusters(result);
```

Comment:
Shouldn't you also try to extract the `modularity`(cugraph_hierarchical_clustering_result_get_modularity) even if you don't test it to make sure nothing is wrong with that (similar the SG CAPI test)?